src/cmd/cmd.zig

```diff
@@ -618,23 +617,53 @@ fn getMyDataFromIpEcho(
 }
 
 fn getEntrypoints(logger: Logger) !std.ArrayList(SocketAddr) {
+    const EntrypointSet = std.ArrayHashMap(void, void, struct {
+        // zig fmt: off
+        pub fn hash(_: @This(), _: void) u32 { unreachable; }
+        pub fn eql(_: @This(), _: void, _: void, _: usize) bool { unreachable; }
+        // zig fmt: on
+    }, true);
+    const EntrypointCtx = struct {
+        entrypoints: []const SocketAddr,
+        pub fn hash(_: @This(), entrypoint: SocketAddr) u32 {
+            const array, const len = entrypoint.toString();
+            return std.array_hash_map.hashString(array[0..len]);
+        }
+        pub fn eql(ctx: @This(), a: SocketAddr, _: void, b_index: usize) bool {
+            return a.eql(&ctx.entrypoints[b_index]);
+        }
+    };
```

Comment:
this seems like overkill for something thats not performance critical - can you just use key to void approach with autocontext?

Reply:
I think this is a fairly reasonable approach, it's a simple context that's re-using the items already stored in the existing array. Why duplicate state?

Reply:
f761151

Reply:
the main problem is it adds unnecessary complexity which reduces the code readability - our first priority is simplicity and easy to ready code

the code block is only run once on startup so a small amount of state duplication is fine

Reply:
Alright I suppose, do you want to keep the context for possible future use, or should I remove it, and we can add it back on a needs-to-be basis?

Reply:
7430655d36eca21cf19986cf29dd5aaaf8e811ee